sdk/inc/azure/core/az_mqtt5_rpc_client_codec.h

```diff
@@ -0,0 +1,237 @@
+ * the response.
+ *
+ * @param[in] client The #az_mqtt5_rpc_client_codec to use.
+ * @param[out] mqtt_topic A buffer with sufficient capacity to hold the MQTT topic filter. If
+ * successful, contains a null-terminated string with the topic filter that needs to be passed to
+ * the MQTT client.
+ * @param[in] mqtt_topic_size The size, in bytes of \p mqtt_topic.
+ * @param[out] out_mqtt_topic_length __[nullable]__ Contains the string length, in bytes, of \p
+ * mqtt_topic. Can be `NULL`.
+ * @pre \p client must not be `NULL`.
+ * @pre \p mqtt_topic must not be `NULL`.
+ * @pre \p mqtt_topic_size must be greater than 0.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The topic was created successfully.
+ * @retval #AZ_ERROR_NOT_ENOUGH_SPACE The buffer is too small.
+ */
+AZ_NODISCARD az_result az_mqtt5_rpc_client_codec_get_subscribe_topic(
+    az_mqtt5_rpc_client_codec* client,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length);
+
+/**
+ * @brief Attempts to parse a received message's topic.
+ *
+ * @param[in] client The #az_mqtt5_rpc_client_codec to use.
+ * @param[in] received_topic An #az_span containing the received topic.
+ * @param[out] out_response If the message is meant for this client, contains the executor id and
+ * command name of the message.
+ * @pre \p client must not be `NULL`.
+ * @pre \p received_topic must be a valid #az_span of size greater than or equal to 1.
+ * @pre \p out_response must not be `NULL`.
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The topic was parsed successfully.
+ * @retval #AZ_ERROR_IOT_TOPIC_NO_MATCH If the topic is not matching the expected format or is not
+ * meant for this client.
+ *
+ */
+AZ_NODISCARD az_result az_mqtt5_rpc_client_codec_parse_received_topic(
+    az_mqtt5_rpc_client_codec* client,
+    az_span received_topic,
+    az_mqtt5_rpc_client_codec_request_response* out_response);
+
+/**
+ * @brief Initializes an MQTT5 RPC Client Codec.
+ *
+ * @param[out] client The #az_mqtt5_rpc_client_codec to initialize.
+ * @param[in] client_id The client id to use for the response topic.
+ * @param[in] model_id The model id to use for the topics.
+ * @param[in] command_name The command name to use for the topics, or AZ_SPAN_EMPTY to specify per
+ * invocation.
+ * @param[in] options #az_mqtt5_rpc_client_codec_options to use for the RPC Client or NULL to use
+ * the defaults.
+ * @pre \p client must not be `NULL`.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The client was initialized successfully.
+ * @retval #AZ_ERROR_ARG An invalid argument was provided.
+ */
+AZ_NODISCARD az_result az_mqtt5_rpc_client_codec_init(
+    az_mqtt5_rpc_client_codec* client,
+    az_span client_id,
+    az_span model_id,
+    az_span command_name,
```

Comment:
Same as above: command_name should be a variable passed when requesting and parsed from a response.